lib/modules/dosomething/dosomething_rogue/dosomething_rogue.cron.inc

```diff
@@ -53,6 +53,23 @@ function dosomething_rogue_retry_failed_reportbacks() {
             ->condition('id', $task->id)
             ->execute();
         }
+      } elseif ($task->type === 'signup') {
+        $data = (array)$task;
```

Comment:
I think if you cast this to an array here, you should use that array throughout the logic. It is a little hard to understand why you are using the `$task` object some places and the `$data` array in others.